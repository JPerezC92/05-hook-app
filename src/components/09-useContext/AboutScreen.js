import React, { useContext } from "react";
import UserContext from "./UserContext";

const AboutScreen = () => {
  const { setUser } = useContext(UserContext);

  const handleLogout = () => setUser({});

  return (
    <div>
      <h1>AboutScreen</h1>
      <hr />

      <button className="btn btn-warning" onClick={handleLogout}>
        Logout
      </button>
    </div>
  );
};

export default AboutScreen;
