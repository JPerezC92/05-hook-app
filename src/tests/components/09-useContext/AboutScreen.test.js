import { mount } from "enzyme";
import AboutScreen from "../../../components/09-useContext/AboutScreen";
import UserContext from "../../../components/09-useContext/UserContext";

describe("Pruebas en el componente <AboutScreen />", () => {
  const setUser = jest.fn();

  const wrapper = mount(
    <UserContext.Provider value={{ setUser }}>
      <AboutScreen />
    </UserContext.Provider>
  );

  test("Debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de llamarse setUser con el argumento esperado", () => {
    wrapper.find("button").simulate("click");

    expect(setUser).toHaveBeenCalledTimes(1);
    expect(setUser).toHaveBeenCalledWith({});
  });
});
