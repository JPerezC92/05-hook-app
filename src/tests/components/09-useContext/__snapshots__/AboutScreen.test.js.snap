// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AboutScreen /> Debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    AboutScreen
  </h1>
  <hr />
  <button
    className="btn btn-warning"
    onClick={[Function]}
  >
    Logout
  </button>
</div>
`;
